include/onnxruntime/core/session/onnxruntime_cxx_api.h

```diff
@@ -196,46 +205,20 @@ struct Base {
   Base(const Base&) = delete;
   Base& operator=(const Base&) = delete;
   Base(Base&& v) noexcept : p_{v.p_} { v.p_ = nullptr; }
-  void operator=(Base&& v) noexcept {
-    OrtRelease(p_);
-    p_ = v.p_;
-    v.p_ = nullptr;
-  }
+  void operator=(Base&& v) noexcept { OrtRelease(p_); p_ = v.release(); }
```

Comment:
>void [](http://example.com/codeflow?start=2&length=4)

is there a reason for returning `void` instead of `Base&`?

Reply:
It's only used internally, so nobody should ever be chaining these. It's a protected method.